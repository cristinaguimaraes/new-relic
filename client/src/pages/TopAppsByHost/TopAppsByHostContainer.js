import { TopAppsByHost } from './TopAppsByHost';
import {withRouter} from "react-router-dom";
import {connect} from "react-redux";
import {hostsWithApps} from "../../redux/reducers/apps";

const mapStateToProps = state => ({
    data: hostsWithApps(state.apps)
});

export const TopAppsByHostContainer = withRouter(
    connect(
        mapStateToProps,
        null
    )(TopAppsByHost)
);


