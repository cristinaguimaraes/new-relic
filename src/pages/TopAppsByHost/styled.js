import styled from "styled-components";

export const Content = styled.div`
  width: 840px;
  display: flex;
  justify-content: center;
  padding: 30px;
  flex-direction: column;
  background-color: #f8f8f8;
`;

export const Header = styled.div`
  font-size: 35px;
  font-family: "Helvetica Bold";
  color: #000000;
  line-height: 43px;
`;

export const Text = styled.div`
  font-family: "Helvetica Bold";
  font-size: 25px;
  margin-top: 15px;
`;
