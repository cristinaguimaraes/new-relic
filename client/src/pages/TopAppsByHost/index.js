import { TopAppsByHostContainer } from './TopAppsByHostContainer';
export { TopAppsByHostContainer as TopAppsByHost };