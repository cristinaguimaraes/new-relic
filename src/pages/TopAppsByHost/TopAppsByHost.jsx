import React from "react";
import { Card } from "components/Card";
import { Content, Header, Text } from "./styled";
import { arrayOf, number, shape, string } from "prop-types";
import { Redirect } from "react-router-dom";

const TopAppsByHost = ({ data }) => {
  const [, , id] = window.location.pathname.split("/");

  if (!data || Object.keys(data).length === 0) return <Redirect to="/" />;

  return (
    <Content>
      {data[id] ? (
        <>
          <Header>Top 25 Apps</Header>
          <Card host={id} apps={data[id].slice(0, 25)} />
        </>
      ) : (
        <Text>Sorry, this host is not in the list!</Text>
      )}
    </Content>
  );
};

TopAppsByHost.propTypes = {
  data: shape({
    string: arrayOf(
      shape({
        host: arrayOf(string).isRequired,
        name: string.isRequired,
        apdex: number.isRequired,
        version: number.isRequired
      })
    )
  }).isRequired
};

export { TopAppsByHost };
