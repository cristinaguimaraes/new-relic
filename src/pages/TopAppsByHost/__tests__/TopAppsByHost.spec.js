import { TopAppsByHost } from "../TopAppsByHost";
import { shallow } from "enzyme";
import React from "react";

describe("TopAppsByHost component", () => {
  test("should render properly", () => {
    expect(
      shallow(
        <TopAppsByHost
          data={{
            "92116865-5462.conor.com": [
              {
                name: "Small Fresh Pants - Kautzer - Boyer, and Sons",
                contributors: ["Edwin Reinger", "Ofelia Dickens"],
                version: 7,
                apdex: 68
              }
            ]
          }}
        >
          Test
        </TopAppsByHost>
      )
    ).toMatchSnapshot();
  });
});
