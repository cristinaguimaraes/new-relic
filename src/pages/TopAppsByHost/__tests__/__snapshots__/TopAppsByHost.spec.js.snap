// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopAppsByHost component should render properly 1`] = `
<styled.div>
  <styled.div>
    Sorry, this host is not in the list!
  </styled.div>
</styled.div>
`;
